.github/workflows/deploy.yml

```diff
@@ -53,23 +59,39 @@ jobs:
       - name: Build and push docker image from builder target
         uses: docker/build-push-action@v2
         with:
-          build-args: BUILDKIT_INLINE_CACHE=1
+          build-args: |
+            BUILDKIT_INLINE_CACHE=1
+            VERSION=${{ env.TAG }}
           cache-from:
-            ${{ env.DOCKERHUB_REPOSITORY }}:develop
+            ${{ env.DOCKERHUB_REPOSITORY_BASE }}${{ github.event.inputs.environment }}:${{ github.event.inputs.environment }}
```

Comment:
Hmm, I was thinking the tag would correspond to git ref rather than environment. After all, we know which environment it is from repository name.

Reply:
No, we want to cache from the environment here, as before. We then tag is as both e.g. `dev` and the commit hash. That way, we can deploy the commit hash, but use `dev` as a starting point for the next image build

Reply:
Happy to chat about this if that didn't make sense